directory/directory_dest.go

```diff
@@ -41,7 +41,7 @@ func (d *dirImageDestination) SupportsSignatures() error {
 
 // ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination.
 func (d *dirImageDestination) ShouldCompressLayers() bool {
-	return false
+	return d.ref.compressed
```

Comment:
_Really_ not a fan of `dir:` modifying the blobs, because of the the “debugging” / “just tell me what is in the source” purposes.